src/line_absorption.jl

```diff
@@ -30,6 +31,12 @@ function line_absorption!(α, linelist, λs, temp, nₑ, n_densities, partition_
     lb = 1
     ub = 1
     β = @. 1/(kboltz_eV * temp)
+
+    # precompute number density / partition function for each species in the linelist
+    n_div_Z = map(collect(Set([l.species for l in linelist]))) do spec
+        spec => @. (n_densities[spec] / partition_fns[spec](log(temp)))
+    end |> Dict
```

Comment:
Do we actually save much time from computing this ahead of time? (if so, is it related to hidden auto-diff calculations?)

Reply:
I tested this to make sure we saved time in normal cases.  It's not a breakthrough or anything but it is faster.  This was actually motivated by performance alone.  It's not necessary for autodiff to work.  Sorry, this is a bit of a frankenstein PR.